src/acvp_build_register.c

```diff
@@ -105,13 +105,16 @@ static ACVP_RESULT acvp_build_hash_register_cap(JSON_Object *cap_obj, ACVP_CAPS_
     JSON_Value *msg_val = NULL;
     JSON_Object *msg_obj = NULL;
     ACVP_HASH_CAP *hash_cap = cap_entry->cap.hash_cap;
+    const char *revision = NULL;
 
     if (!hash_cap) {
         return ACVP_MISSING_ARG;
     }
 
     json_object_set_string(cap_obj, "algorithm", acvp_lookup_cipher_name(cap_entry->cipher));
-    json_object_set_string(cap_obj, "revision", ACVP_SHA_REVISION);
+
+    revision = acvp_lookup_cipher_revision(cap_entry->cipher);
```

Comment:
Do we want to generate some type of error generated when we get a NULL ?
Maybe just do it in the lookup function ?

Reply:
Ah good catch